Add plugin print css

mmcfarland opened the pull request:
Plugins can specify the css that they want enabled during plugin print.

##### To Test
* Open the Identify Point sample plugin, get passed the Info Graphic screen and click print
* The print preview should show the info graphic
* Open Split View and have both maps have an active Identify Point plugin
* Print one of the two plugins and ensure that only one is print previewed

Note that I didn't spend much time on actually positioning the graphic for print nicely, just demonstrating that plugins have the ability to do so.

Only the last three commits are relevant to this PR

Connects #426

kdeloach commented on src/GeositeFramework/js/Plugin.js:
```diff
@@ -417,6 +445,13 @@ require(['use!Geosite',
+                href: path,
+                'class': className
+            }).appendTo('head');
+    }
```
Needs a tab here.

kdeloach commented:
The map disappears if you follow these steps:
1. Switch to Map 2 (after page loads)
2. Open Identify Point plugin and click print icon
3. Switch to Map 1

kdeloach commented:
Not sure if this is intentional, but the plugin text does not appear in the print out.

![image](https://cloud.githubusercontent.com/assets/43062/10441953/68625ad4-711c-11e5-8cac-3da139d69801.png)

![image](https://cloud.githubusercontent.com/assets/43062/10441955/6a27980c-711c-11e5-9b33-2abe584d9021.png)

mmcfarland commented:
Looking into the disappearing map, good catch.

mmcfarland commented:
Sadly, the map disappears if you follow the same sequence but just do a Ctrl-P to bring the print dialog up and not interact with any of the new code or plugins.

kdeloach commented:
Will test to see if this issue exists on master.

kdeloach commented:
Confirmed this issue is present on master. I'll open an issue.

kdeloach commented:
+1

caseypt commented:
After reviewing this, this comment may be more relevant here: https://github.com/CoastalResilienceNetwork/GeositeFramework/pull/429#issuecomment-147723621.

Do we want the selected location in the print preview?

caseypt commented:
When printing with split screen maps enabled, the infographic is cut off. Not sure what the solution is for this.

Map 1:
![screen shot 2015-10-13 at 10 19 06 am](https://cloud.githubusercontent.com/assets/1042475/10457343/e8f7d63e-7193-11e5-9bd8-73a44edbf940.png)

Map 2:
![screen shot 2015-10-13 at 10 19 52 am](https://cloud.githubusercontent.com/assets/1042475/10457353/f618c5da-7193-11e5-9e84-7464e81edb14.png)

caseypt commented:
Regarding ^, maybe, in addition to hiding the second map, expand the width of printed map?

mmcfarland commented:
Sorry, I should have been more clear.  I'm just using the Sample Plugin to test the workflow, we're not on the hook for it to print anything reasonable and it's not deployed anywhere.  I didn't really spend any time on the print.css for this plugin to make the output look good, mostly just looking to see that it is loaded and printable.

mmcfarland commented:
Although the pane width issue may be a big deal.  I'll try adjusting the pane width in the framework print css as you suggest

mmcfarland commented:
Updated whitespace, comments and split screen print width in https://github.com/mmcfarland/GeositeFramework/commit/d0660020113da45d4f628f3f98943f066712bbf4

mmcfarland commented:
Might hold off - there is a difference between real print preview and the emulate CSS media option, which I was using.

mmcfarland commented:
Ok, updated and ready for testing.

caseypt commented:
+1. Looks good. Printing from either map produces a good result.
